LearningMissionSimulation/PlaygroundVahe.cs

```diff
@@ -86,32 +92,72 @@ public void ActivateAccounts()
             }
         }
 
-        public void AssignModulesToInstructors()
+        public void CreateSubjects(int subjectCount)
         {
-            throw new NotImplementedException();
+            for (int i = 0; i < subjectCount; i++)
+            { 
+                Subject subject = ObjectGenerator.GenerateSubject();
+                subjectList.Add(subject);
+                subjectIdList.Add(subject.Id);
+            }
         }
 
-        public void AssignModulesToStudents()
+        public void CreateModules(int moduleCount)
         {
-            throw new NotImplementedException();
+            for (int i = 0;  i < moduleCount; i++)
+            {
+                Guid subjectId = subjectIdList[AttributeGenerator.random.Next(0, subjectIdList.Count)];
+                Module module = ObjectGenerator.GenerateModule(subjectId);
+                moduleDictionary.Add(module.Id, module);
+                moduleIdList.Add(module.Id);
+            }
         }
 
-        public void CreateClassrooms(int classroomCount)
+        public void AssignModulesToInstructors()
         {
-            throw new NotImplementedException();
+            foreach (var instructor in instructorList)
+            {
+                instructor.ModuleList = GetModuleList();
+            }
         }
 
-        public void CreateModules(int moduleCount)
+        public void AssignModulesToStudents()
         {
-            throw new NotImplementedException();
+            foreach (var student in studentList)
+            {
+                student.CompletedModuleList = GetModuleList();
+            }
         }
 
-        public void RegisterStudentsForClasses()
+        List<Module> GetModuleList()
+        {
+            List<Module> moduleList = new List<Module>();
+            int moduleCount = moduleIdList.Count;
+            int maxModuleCountLimit = 7;
+            int minModuleCountLimit = 1;
+            maxModuleCountLimit = Math.Min(moduleCount, maxModuleCountLimit);
+            minModuleCountLimit = Math.Min(moduleCount, minModuleCountLimit);
+            int count = AttributeGenerator.random.Next(minModuleCountLimit, maxModuleCountLimit);
+
+            for (int i = 0; i < moduleCount; i++)
+            {
+                Module module;
+                Guid id = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
+                moduleDictionary.TryGetValue(id, out module);
+                if (!moduleList.Contains(module))
```

Comment:
if (!moduleList.Contains(module)) will work but it means every time you call it it will be looping through the whole list.  Our list is small and you will not see any performance problems here, but for big collections it could be a problem. Please consider other data structure for the job.